posthog/hogql_queries/legacy_compatibility/feature_flag.py

```diff
@@ -0,0 +1,24 @@
+import posthoganalytics
+from django.conf import settings
+from posthog.cloud_utils import is_cloud
+from posthog.models.user import User
+
+
+def hogql_insights_enabled(user: User) -> bool:
```

Comment:
types can help somewhat here:

```suggestion
def hogql_insights_enabled(user: User | AnonymousUser) -> bool:
```